### PR TITLE
Service host should return 503 on shutdown

### PR DESCRIPTION
**Pull request by EranOfer**

### CODE REVIEW

File: Gigya.Microdot.Orleans.Ninject.Host/MicrodotOrleansServiceHost.cs [hunk 1]
```diff
@@ -168,6 +168,13 @@ protected virtual void PreConfigure(IKernel kernel)
         /// </summary>        
         protected override void OnStop()
         {
+            if (Arguments.BeforeStopSet503WaitTime.HasValue)
+            {
+                Kernel.Get<IServcieShutdownSource>().Shutdown();
+                Task.Delay(Arguments.BeforeStopSet503WaitTime.Value).Wait();
+                Task.Delay(Arguments.BeforeStopSet503WaitTime.Value).Wait();
```
Comment (bronsh) [inline Gigya.Microdot.Orleans.Ninject.Host/MicrodotOrleansServiceHost.cs]: Why twice? 
Is it required to wait double-time? (If so, you can do Task.Delay(Arguments.BeforeStopSet503WaitTime.Value*2).Wait()

File: Gigya.Microdot.SharedLogic/ServiceArguments.cs [hunk 2]
```diff
@@ -70,6 +70,10 @@ public class ServiceArguments
         /// </summary>
         public int? ShutdownWhenPidExits { get; }
 
+        /// <summary>
+        /// Specifies time before stoping  in this time the servcie status will be 503.
+        /// </summary>
+        public TimeSpan? BeforeStopSet503WaitTime { get;  }
```
Comment (bronsh) [inline Gigya.Microdot.SharedLogic/ServiceArguments.cs]: change entry name to WaitTimeBeforeStoppingService
The 503 status code is an implementation detail, not for configuration.

File: Gigya.Microdot.SharedLogic/ShutDownToken.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        CancellationToken Token { get; }
+    }
+
+    internal interface IServcieShutdownSource : IServcieShutdownToken
+    {
+        void Shutdown();
+    }
+
+    internal class ShutdownToken : IServcieShutdownSource
```
Comment (bronsh) [inline Gigya.Microdot.SharedLogic/ShutDownToken.cs]: Change class name to ServiceShutDownToken (to match the interface name)

File: Gigya.Microdot.SharedLogic/ServiceArguments.cs (hunk 2 above)
Comment (EranOfer) [reply Gigya.Microdot.SharedLogic/ServiceArguments.cs]: Used for other purposes.

**Pull request merged**